torchdata/dataloader2/reading_service.py

```diff
@@ -174,38 +188,72 @@ def __init__(
         self.multiprocessing_context = multiprocessing_context
         self.processes = []
         self.datapipes = []
-        self.combined_datapipes = None
+        self.end_datapipe = None
```

Comment:
I change it to `end_datapipe` because we need to store the last DataPipe for both In-process and multiprocessing cases.